thirdparty/runtime_environments.go

```diff
@@ -96,3 +96,49 @@ func (c *RuntimeEnvironmentsClient) getOSInfo(ctx context.Context, amiID string,
 	}
 	return osInfo, nil
 }
+
+// ImageInfo represents information about an image with its AMIID and creation date.
+type ImageInfo struct {
+	AMIID        string
+	CreationDate string
+}
+
+// DistoHistoryFilter represents the filtering arguments for getHistory. The Distro field is required and the other fields are optional.
+type DistroHistoryFilter struct {
+	Distro string
+	Page   int
+	Limit  int
+}
+
+func (c *RuntimeEnvironmentsClient) getHistory(ctx context.Context, opts DistroHistoryFilter) ([]ImageInfo, error) {
+	params := url.Values{}
+	if opts.Distro == "" {
+		return nil, errors.New("no distro provided.")
```

Comment:
nit. no punctuation necessary
```suggestion
		return nil, errors.New("no distro provided")
```